File: project/app/src/main/java/org/owntracks/android/services/worker/MQTTMaybeReconnectAndPingWorker.java
package org.owntracks.android.services.worker;


import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import org.owntracks.android.injection.components.AppComponentProvider;
import org.owntracks.android.services.MessageProcessor;

import javax.inject.Inject;

import timber.log.Timber;

public class MQTTMaybeReconnectAndPingWorker extends Worker {

    @Inject
    MessageProcessor messageProcessor;

    public MQTTMaybeReconnectAndPingWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
        AppComponentProvider.getAppComponent().inject(this);
    }

    @NonNull
    @Override
    public Result doWork() {
        Timber.tag("MQTT").d("MQTTMaybeReconnectAndPingWorker doing work on threadID: %s", Thread.currentThread());
        if (!messageProcessor.isEndpointConfigurationComplete())
            return Result.failure();
        return messageProcessor.statefulReconnectAndSendKeepalive() ? Result.success() : Result.retry();
    }
}

File: project/app/src/main/java/org/owntracks/android/services/worker/MQTTMaybeReconnectAndPingWorker.kt
//package org.owntracks.android.services.worker
//
//import android.content.Context
//import androidx.hilt.work.HiltWorker
//import androidx.work.ListenableWorker
//import androidx.work.Worker
//import androidx.work.WorkerParameters
//import dagger.assisted.Assisted
//import dagger.assisted.AssistedInject
//import org.owntracks.android.services.MessageProcessor
//import timber.log.Timber
//import javax.inject.Inject
//
//@HiltWorker
//class MQTTMaybeReconnectAndPingWorker @AssistedInject constructor(
//    @Assisted context: Context,
//    @Assisted workerParams: WorkerParameters,
//    private val messageProcessor: MessageProcessor
//) : Worker(context, workerParams) {
//    override fun doWork(): Result {
//        Timber.d(
//            "MQTTMaybeReconnectAndPingWorker started on threadID: %s",
//            Thread.currentThread()
//        )
//        if (!messageProcessor.isEndpointReady) return Result.failure()
//        return if (messageProcessor.statefulReconnectAndSendKeepalive()) Result.success() else Result.retry()
//    }
//
//    class Factory @Inject constructor(private val messageProcessor: MessageProcessor) :
//        ChildWorkerFactory {
//        override fun create(appContext: Context, params: WorkerParameters): ListenableWorker =
//            MQTTMaybeReconnectAndPingWorker(appContext, params, messageProcessor)
//    }
//}
